### DISCUSSION TITLE
Get IP address of player

### DISCUSSION THREAD

**Discussion by onLaait** [Minestom/Minestom#1680] [Help] [answered]
Hi, how can i get IP address of player?

**Comment by Moulberry** [comment, marked as answer]
You can get the `PlayerConnection` object from the Player and check if it is a `PlayerSocketConnection`.
If it is, you can call PlayerSocketConnection#getRemoteAddress
If it's a FakePlayerConnection, then there obviously won't be an IP address associated with the Player

Even if you are running the server behind a proxy, the remoteAddress should still point to the "real" connection address (See https://github.com/Minestom/Minestom/blob/993af51556284565ec7ec760df39ae80045eb3c8/src/main/java/net/minestom/server/network/packet/client/login/LoginPluginResponsePacket.java#L66). This may change though, so be warned.
